go-controller/pkg/allocator/udn/allocator.go

```diff
@@ -0,0 +1,50 @@
+package udn
+
+import (
+	"fmt"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+type Allocator struct {
+	watchFactory factory.NodeWatchFactory
+	nodeName     string
+}
+
+func New(watchFactory factory.NodeWatchFactory, nodeName string) *Allocator {
+	return &Allocator{
+		watchFactory: watchFactory,
+		nodeName:     nodeName,
+	}
+}
+
+func (a *Allocator) calculateIDsFromNetwork(idName, networkName string, base, limit, cardinality uint) (uint, error) {
+	if cardinality < 1 {
+		return 0, fmt.Errorf("invalid arguments, cardinality '%d' has to be bigger than '1'", cardinality)
+	}
+	if limit <= base {
+		return 0, fmt.Errorf("invalid arguments, limit '%d' has to be bigger than base '%d'", limit, base)
+	}
+	node, err := a.watchFactory.GetNode(a.nodeName)
+	if err != nil {
+		return 0, fmt.Errorf("failed geting node to retrieve network-ids at node '%s' when calculating '%s' for network '%s': %w", a.nodeName, idName, networkName, err)
+	}
+	networkIDByNetworkName, err := util.GetNodeNetworkIDsAnnotationNetworkIDs(node)
+	if err != nil {
+		return 0, fmt.Errorf("failed geting network-ids from node '%s' when calculating '%s' network '%s': %w", a.nodeName, idName, networkName, err)
+	}
+	networkID, ok := networkIDByNetworkName[networkName]
+	if !ok {
+		return 0, fmt.Errorf("missing id for network '%s' when calculating '%s'", networkName, idName)
+	}
+	if networkID < 1 {
+		return 0, fmt.Errorf("invalid arguments, '%s' network-id has to be bigger than '0'", networkName)
+	}
+
+	if base+uint(networkID)*cardinality >= limit {
+		return 0, fmt.Errorf("out of bounds '%s' network id '%d' when calculating '%s'", networkName, networkID, idName)
+	}
+
+	return base + 1 + (cardinality * uint(networkID)) - cardinality, nil
```

Comment:
so what's the max for networkID here?

Reply:
Will dump the overpassed max at the error.